### PR TITLE
Forking workflow directions CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by taylorfturner**

### CODE REVIEW

**Comment by taylorfturner** [comment]
WIP to add image of branching process and viz of merging

**Review by tyfarnan — APPROVED** [review]

**Review by micdavis — CHANGES_REQUESTED** [review]
blocking for now

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -60,6 +60,22 @@ Pull requests are the best way to propose changes to the codebase. We actively w
...
+3. To resolve issues between `main` and your feature branch use interactive rebase. Documentation [here](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow).
+4. Maintainers may from time to time make an exception to the above workflow and host a feature branch on `capitalone/dataprofiler`. Permissions would then be reconfigured for a force push to core team membership exclusively.
+
+See below image for the above text points visualized of both `main` and `gh-pages`.
+
+<p text-align="left">
+    <picture>
+      <source srcset="https://github.com/capitalone/DataProfiler/raw/gh-pages/docs/source/_static/images/branching_workflow_diagram.png">
+    </picture>
+</p>
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply .github/CONTRIBUTING.md:77]: WIP locally

**Review by tyfarnan — DISMISSED** [review]

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -60,6 +60,28 @@ Pull requests are the best way to propose changes to the codebase. We actively w
 5. Make sure your code lints.
 6. Issue that pull request!
 
+## Feature Pull Request Workflow
+When working on a new feature, which will require multiple pull requests, the following workflow is to ensure that it is most efficient and safe for developers and reviewers.
+
+1. A feature branch that an author is working on should live on their fork and not on `capitalone/dataprofiler`. This is becuase a rebase requires a force push and we do not want to open up permissions on a feature branch on `capitalone/dataprofiler` for any potential user to force push to a `feature/<branch_name>` on `capitalone/dataprofiler`.
+2. Feature branch naming convention:
+  - Feature branches on forks and on `capitalone/dataprofiler: `feature/<feature_name>`
```
Comment (micdavis) [inline .github/CONTRIBUTING.md:68]:
```suggestion
  - Feature branches on forks and on `capitalone/dataprofiler`: `feature/<feature_name>`
```

**Review by JGSweets — APPROVED** [review]

**Review by micdavis — APPROVED** [review]

**Pull request merged**